### PR TITLE
Re-attempt to acquire lock upon start-up when possible

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

File: frontend/src/main/scala/bloop/Bloop.scala [hunk 1]
```diff
@@ -85,12 +85,26 @@ object Bloop {
...
         startServer(Left(hostPort))
       case Right(lockFiles) =>
-        Lock.tryAcquire(lockFiles, LockProcess.default) {
-          startServer(Right(lockFiles.socketPaths))
-        } match {
-          case Left(err) => throw new Exception(err)
-          case Right(()) =>
+        val period = 3.second
+
+        def loop(): Unit = {
```
Comment (Gedochao) [inline frontend/src/main/scala/bloop/Bloop.scala:90]:
```suggestion
        @tailrec
        def loop(): Unit = {
```

File: frontend/src/main/scala/bloop/Bloop.scala [hunk 2]
```diff
@@ -85,12 +85,26 @@ object Bloop {
...
+          val res = Lock.tryAcquire(lockFiles, LockProcess.default) {
+            startServer(Right(lockFiles.socketPaths))
+          }
+          res match {
+            case Left(err: LockError.RecoverableError) =>
+              System.err.println(s"Caught $err, trying again in $period")
+              Thread.sleep(period.toMillis)
+              loop()
+            case Left(err: LockError.AlreadyRunning) =>
```
Comment (Gedochao) [inline frontend/src/main/scala/bloop/Bloop.scala:99]:
```suggestion
            case Left(_: LockError.AlreadyRunning) =>
```

File: frontend/src/main/scala/bloop/Bloop.scala [hunk 3]
```diff
@@ -85,12 +85,26 @@ object Bloop {
...
+
+        def loop(): Unit = {
+          val res = Lock.tryAcquire(lockFiles, LockProcess.default) {
+            startServer(Right(lockFiles.socketPaths))
+          }
+          res match {
+            case Left(err: LockError.RecoverableError) =>
+              System.err.println(s"Caught $err, trying again in $period")
+              Thread.sleep(period.toMillis)
+              loop()
```
Comment (Gedochao) [inline frontend/src/main/scala/bloop/Bloop.scala:98]: is it ok to loop infinitely here? or should there be a limit to the number of retries? 🤔

Comment (alexarchambault) [reply frontend/src/main/scala/bloop/Bloop.scala:98]: I need to add a timeout, I thought about it only after…

File: frontend/src/main/scala/bloop/Bloop.scala (hunk 1 above)
Comment (alexarchambault) [reply frontend/src/main/scala/bloop/Bloop.scala]: Should be fixed.

File: frontend/src/main/scala/bloop/Bloop.scala (hunk 2 above)
Comment (alexarchambault) [reply frontend/src/main/scala/bloop/Bloop.scala]: Should be fixed

**Pull request merged**